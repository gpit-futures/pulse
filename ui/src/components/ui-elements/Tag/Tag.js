import React from 'react';

export const Tag = ({ children, style, ...restProps }) => <span className={`label label-${style}`} {...restProps}>{children}</span>;

export const TagList = ({ children }) => <div className="control-group">{children}</div>;
