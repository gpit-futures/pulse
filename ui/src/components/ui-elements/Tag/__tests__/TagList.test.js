import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import { TagList } from '../Tag';

Enzyme.configure({ adapter: new Adapter() });

const testProps = {
  children: 'test-children',
};

describe('Component <TagList />', () => {
  it('should renders correctly', () => {
    const component = shallow(
      <TagList
        children={testProps.children}
      />);

    expect(component.find('.control-group')).toHaveLength(1);
    expect(component.find('.control-group').text()).toEqual(testProps.children);

    expect(component).toMatchSnapshot();
  });
});

