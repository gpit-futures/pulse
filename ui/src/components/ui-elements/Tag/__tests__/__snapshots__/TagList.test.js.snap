// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <TagList /> should renders correctly 1`] = `
<div
  className="control-group"
>
  test-children
</div>
`;
